cmd/eksctl-anywhere/cmd/options.go

```diff
@@ -108,3 +114,49 @@ func markFlagHidden(cmd *cobra.Command, flagName string) {
 		logger.V(5).Info("Warning: Failed to mark flag as hidden: " + flagName)
 	}
 }
+
+func buildCliConfig(clusterSpec *cluster.Spec) *config.CliConfig {
+	cliConfig := &config.CliConfig{}
+	if clusterSpec.FluxConfig != nil && clusterSpec.FluxConfig.Spec.Git != nil {
+		cliConfig.GitSshKeyPassphrase = os.Getenv(config.EksaGitPassphraseTokenEnv)
+		cliConfig.GitPrivateKeyFile = os.Getenv(config.EksaGitPrivateKeyTokenEnv)
+		cliConfig.GitKnownHostsFile = os.Getenv(config.EksaGitKnownHostsFileEnv)
+	}
+
+	return cliConfig
+}
+
+func (c *clusterOptions) directoriesToMount(clusterSpec *cluster.Spec, cliConfig *config.CliConfig) ([]string, error) {
+	dirs := c.mountDirs()
+	fluxConfig := clusterSpec.FluxConfig
+	if fluxConfig != nil && fluxConfig.Spec.Git != nil {
+		dirs = append(dirs, filepath.Dir(cliConfig.GitPrivateKeyFile))
+		dirs = append(dirs, filepath.Dir(cliConfig.GitKnownHostsFile))
+		if c.installPackages != "" {
+			dirs = append(dirs, filepath.Dir(c.installPackages))
+		}
```

Comment:
These lines have been added from the original - whats the motivation here? I'm also struggling some to understand why we conditionally append the base path of `installPackages` on flux configuration being present?

Reply:
I just added the if statement for installPackages. Previously delete and upgrade commands just called the createClusterOptions installPackages directory. I'll see if this can be split out and if it is still necessary for createCluster.

Reply:
Followed up with Abdul, you're right the installPackages mount can be moved out of the flux if statement. Updated the function to accept additional directories to mount, `create cluster` will pass c.installPackages as an additional directory.